mule-extensions-api/src/main/java/org/mule/runtime/extension/api/component/value/ValueDeclarer.java

```diff
@@ -35,6 +37,14 @@ public interface ValueDeclarer {
    */
   void arrayValue(Consumer<ArrayValueDeclarer> arrayValueDeclarerConsumer);
 
+  /**
+   * Declares that the described value represents a {@link org.mule.runtime.api.config.PoolingProfile}
+   *
+   * @param poolingProfileValueDeclarerConsumer a consumer to configure the value of the pooling profile
+   * @return the declarer to be used to declare the pooling profile value
+   */
+  void poolingProfile(Consumer<PoolingProfileValueDeclarer> poolingProfileValueDeclarerConsumer);
```

Comment:
weird to have this at the value declarer level since this is only applicable at the connection provider level. If the purpose is to mimic what MF has, then why not bringing the entire constructs and create de concept of a config and connection configurers, etc... Putting this at this value seems like mixing concerns

Reply:
I get what you say, it's confusing that a user is defining a parameter for an operation can configure its value as a pooling profile. 

What it would enable is, if there is a connector that defines a parameter of an infrastructure type, or a pojo that has a field of one of those types, define its value. I get that this case may not be ideal and if we chose to not support it we can move forward with a definition similar to what mule-framework has defined.

I will change this draft creating a ConnectionProviderParameterization that extends from ComponentParameterization and can configure the PoolingProfile.